updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,76 @@
+
+To prevent specific nodes from being updated, and thus, not drained, cordoned, and updated, you can create custom MCPs to ensure that the nodes are not updated. The MCO does not update any paused MCPs. You can create one or more custom MCPs, which can give more control over the sequence in which nodes are updated. After the update of the first MCP, the application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+In case of a failure, such as your applications not working on the updated nodes, you can cordon and drain the nodes in the pool, which moves the application pods to other nodes to help maintain the quality-of-service for the applications.
+
+[NOTE]
+====
+To ensure that the cluster remains stable, creating a custom MCP for the control plane nodes (also known as the master nodes) is not supported. The Machine Config Operator (MCO) ignores any custom MCP created for the control plane nodes.
+====
+
+An {product-title} cluster is highly-available by design. There are many features (such as pod disruption budgets, pod affinity, health checks, replicas, and so forth) that  make your applications highly-available in case of node failure. However, there can be scenarios where a slow rollout of a new {product-title} version, also known as a _canary_ release, to worker nodes is preferred.
+
+You can use this IPSU process with all documented {product-title} update processes, but does not update Red Hat Enterprise Linux (RHEL) machines.
+
+include::modules/update-using-custom-machine-config-pools-about.adoc[leveloffset=+1]
+include::modules/update-using-custom-machine-config-pools-mcp.adoc[leveloffset=+1]
+include::modules/update-using-custom-machine-config-pools-pause.adoc[leveloffset=+1]
+
+[id="update-using-custom-machine-config-pools-update_{context}"]
+== Performing the cluster update
+
+When the MCPs enter ready state, you can peform the update. See one of the following update methods, as appropriate for your clustre:
+
+* xref:../updating/updating-cluster-between-minor.adoc#update-upgrading-web_updating-cluster-between-minor[Updating a cluster between minor versions]
+* xref:../updating/updating-cluster.adoc#update-upgrading-web_updating-cluster[Updating a cluster within a minor version from the web console]
+* xref:../updating/updating-cluster-cli.adoc#update-upgrading-cli_updating-cluster-cli[Updating a cluster within a minor version by using the CLI]
+
+After the update is complete, you need to remove the nodes from the custom MCPs, as shown in the following setion.
```

Comment:
We do not need to remove the labels as there is no harm of having the machine config pools if they are not paused. Having the machine config pools will help them when they will update the cluster in future. However we need to provide the steps to remove the labels in the doc from the doc completion point of view. Also may be customers want to remove the labels for some reason.